lib/components/Button/Button.tsx

```diff
@@ -1,33 +1,121 @@
   /** Callback fired on tap/click of the button */
   onClick: () => void;
-  /** Variant of the button */
-  variant?: ButtonVariant;
+  /** Size of the button */
+  size?: 'sm' | 'md' | 'lg';
+  /** Button variant */
+  variant?: 'gray' | 'white' | 'delete';
   /** Button disabled for any actions */
   disabled?: boolean;
+  /** Icon shown on the button */
+  icon?: string;
+  /** Aria label for the icon */
+  iconAriaLabel?: string;
+  /** Icon side */
+  iconSide?: 'left' | 'right';
 }
 
-export const Button = ({ label, onClick, variant = 'base', disabled = false }: ButtonProps) => {
-  const className = `m-2 px-4 py-2 font-bold
-      ${variant === 'base' ? 'bg-jod-base text-jod-white' : ''}
-      ${variant === 'primary' ? 'bg-jod-primary text-jod-white' : ''}
-      ${variant === 'outlined' ? 'border border-jod-black text-jod-black' : ''}
-      ${variant === 'text' ? 'text-jod-black' : ''}
-      ${disabled ? 'cursor-not-allowed opacity-50' : ''}`
+const getSizeClassName = (size: ButtonProps['size']) => {
+  switch (size) {
+    case 'sm':
+      return 'text-button-sm px-6';
+    case 'md':
+      return 'text-button-md px-6';
+    case 'lg':
+      return 'text-button-lg px-[42px]';
+    default:
+      return '';
+  }
+};
+
+const getIconClassName = (size: ButtonProps['size'], leftIcon: boolean, rightIcon: boolean) => {
+  return `
+    ${size === 'sm' && leftIcon ? 'pl-4' : ''}
+    ${size === 'md' && leftIcon ? 'pl-4' : ''}
+    ${size === 'lg' && leftIcon ? 'pl-[21px]' : ''}
+    ${size === 'sm' && rightIcon ? 'pr-4' : ''}
+    ${size === 'md' && rightIcon ? 'pr-4' : ''}
+    ${size === 'lg' && rightIcon ? 'pr-[21px]' : ''}
+  `;
+};
+
+const getVariantClassName = (variant: ButtonProps['variant']) => {
+  switch (variant) {
+    case 'gray':
+      return 'text-primary-gray bg-bg-gray';
+    case 'white':
+      return 'text-primary-gray bg-white';
+    case 'delete':
+      return 'text-alert hover:text-alert focus-visible:text-alert active:text-alert bg-white focus-visible:outline-0 active:bg-white';
+    default:
+      return '';
+  }
+};
+
+const getDisabledClassName = (disabled: ButtonProps['disabled'], variant: ButtonProps['variant']) => {
+  return disabled === false && (variant === 'gray' || variant === 'white')
+    ? 'hover:text-accent focus-visible:text-accent active:bg-accent focus-visible:outline-accent focus-visible:outline focus-visible:outline-[3px] focus-visible:outline-offset-[1.5px] active:text-white'
+    : 'disabled:text-inactive-gray disabled:cursor-not-allowed';
+};
+
+const getButtonClassName = ({
+  size,
+  variant,
+  leftIcon,
+  rightIcon,
+  disabled,
+}: Partial<ButtonProps> & { leftIcon: boolean; rightIcon: boolean }) => {
+  return `flex items-center gap-4 rounded-[30px] select-none
+    ${getSizeClassName(size)}
+    ${getIconClassName(size, leftIcon, rightIcon)}
+    ${getVariantClassName(variant)}
+    ${getDisabledClassName(disabled, variant)}
+  `
+    .replace(/\s+/g, ' ')
+    .trim();
```

Comment:
Maybe there is now enough use that we can have some own util to do these replaces and trims?

Reply:
I created https://jira.eduuni.fi/browse/OPHJOD-325 ticket for that.